lib/spack/spack/build_environment.py

```diff
@@ -275,18 +276,25 @@ def set_build_environment_variables(pkg, env, dirty):
     for dep in link_deps:
         if is_system_path(dep.prefix):
             continue
-        # TODO: packages with alternative implementations of .libs which
-        # are external may place libraries in nonstandard directories, so
-        # there should be a check for that
         query = pkg.spec[dep.name]
+        dep_link_dirs = list()
         try:
-            dep_link_dirs = list(query.libs.directories)
-            link_dirs.extend(dep_link_dirs)
-            if dep in rpath_deps:
-                rpath_dirs.extend(dep_link_dirs)
+            # TODO: packages with alternative implementations of .libs which
```

Comment:
out of curiosity, could you please elaborate? If custom `libs()` contains system libraries (like `intel-mkl`), then we are ok as we filter it. What can one check?

Reply:
Now that you bring it up and I consider it more I think this comment isn't very useful, so I'm going to remove it. There are two issues but both of them can be handled

* `.libs` can be implemented by a package to look in alternative directories (so when there is an external, the Package can know where to look, kind of like how `Python` locates the Python exe
* `filter_system_paths` can remove paths that shouldn't be present